tests/test_georaster.py

```diff
@@ -26,6 +26,46 @@
 DO_PLOT = False
 
 
+def run_gdal_proximity(
+    input_raster: gu.Raster, target_values: list[float] | None, distunits: str = "GEO"
+) -> np.ndarray:
+    """Run GDAL's ComputeProximity and return the read numpy array."""
+    # Rasterio strongly recommends against importing gdal along rio, so this is done here instead.
+    from osgeo import gdal, gdalconst
+
+    # Initiate empty GDAL raster for proximity output
+    drv = gdal.GetDriverByName("MEM")
+    proxy_ds = drv.Create("", input_raster.shape[1], input_raster.shape[0], 1, gdal.GetDataTypeByName("Float32"))
+    proxy_ds.GetRasterBand(1).SetNoDataValue(-9999)
+
+    # Save input in temporary file to read with GDAL
+    # (avoids the nightmare of setting nodata, transform, crs in GDAL format...)
+    temp_dir = tempfile.TemporaryDirectory()
+    temp_path = os.path.join(temp_dir.name, "input.tif")
+    input_raster.save(temp_path)
+    ds_raster_in = gdal.Open(temp_path, gdalconst.GA_ReadOnly)
+
+    # Define GDAL options
+    proximity_options = ["DISTUNITS=" + distunits]
+    if target_values is not None:
+        proximity_options.insert(0, "VALUES=" + ",".join([str(tgt) for tgt in target_values]))
+
+    # Compute proximity
+    gdal.ComputeProximity(ds_raster_in.GetRasterBand(1), proxy_ds.GetRasterBand(1), proximity_options)
+    # Save array
+    proxy_array = proxy_ds.GetRasterBand(1).ReadAsArray().astype("float32")
+    proxy_array[proxy_array == -9999] = np.nan
+
+    # Close GDAL datasets
+    proxy_ds = None
+    ds_raster_in = None
+
+    # Clean temporary directory
+    temp_dir.cleanup()
```

Comment:
This is not required if used with a context manager (my suggestion above)

Reply:
Removed